noir-projects/noir-contracts/contracts/token_contract/src/types/token_note.nr

```diff
@@ -60,16 +60,20 @@ impl NoteInterface<TOKEN_NOTE_LEN, TOKEN_NOTE_BYTES_LEN> for TokenNote {
 
     // TODO(#7738): Nuke this function and have it auto-generated by macros
     fn compute_note_hiding_point(self) -> Point {
+        assert(self.header.storage_slot != 0, "Storage slot must be set before computing note hiding point");
```

Comment:
I placed the assert here to sanity check that everywhere in the codebase the slot is assigned before we try to compute note hiding point. In plenty of the state vars we have the slot check inserted as well.

I think it would make sense to keep the slot check in all the notes and nuke it from the state vars. Does the reviewer agree?

@LHerskind I remember you once explained me why the 0 slot was not working. I think it had something to do with pedersen but I can't exactly remember now. Do you still remember what was the cause? Maybe we can nuke the checks altogether now that pedersen is not used to compute note hashes.